src/transaction/transaction_manager.cpp

```diff
@@ -167,6 +185,18 @@ TransactionQueue TransactionManager::CompletedTransactionsForGC() {
   return hand_to_gc;
 }
 
+void TransactionManager::DeferAction(Action a) {
+  common::SpinLatch::ScopedSpinLatch guard(&deferred_actions_latch_);
+  deferred_actions_.push({time_.load(), a});
+}
+
+ActionQueue TransactionManager::DeferredActionsForGC() {
+  common::SpinLatch::ScopedSpinLatch guard(&deferred_actions_latch_);
+  ActionQueue hand_to_gc(std::move(deferred_actions_));
+  TERRIER_ASSERT(deferred_actions_.empty(), "TransactionManager's queue should now be empty.");
+  return hand_to_gc;
```

Comment:
I think we just want to `return std::move(deferred_actions_)` and lose the `TERRIER_ASSERT`. We can update the `CompletedTransactionsForGC()` method at the same time, since it uses the same logic. The return type will probably have to be changed to r-value reference. If the goal was to avoid copies, we failed. :)